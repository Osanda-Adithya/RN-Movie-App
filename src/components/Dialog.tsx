import React, { FC } from "react";
import { View, StyleSheet, Dimensions, Text } from 'react-native'
import Icon from 'react-native-vector-icons/Ionicons'

type DialogProps = {
    title: string;
    description: string;
}

const { width, height } = Dimensions.get('window');

const Dialog: FC<DialogProps> = ({ title, description }) => {
    return (
        <View style={styles.containerBackground}>
            <View style={styles.constainer}>
                <View style={styles.iconContainer}>
                    <Icon name="warning" color={'red'} size={26} />
                </View>
                <Text style={styles.headerTitle}>{title}</Text>
                <Text style={styles.description}>{description}</Text>
            </View>
        </View>
    )
}

export default Dialog;

const styles = StyleSheet.create({
    constainer: {
        width: width - 50,
        height: 200,
        backgroundColor: 'white',
        borderRadius: 10,
        alignItems: 'center',
        justifyContent: 'center'
    },
    containerBackground: {
        width: width,
        height: height,
        position: 'absolute',
        backgroundColor: 'rgba(4, 36, 68, 0.8)',
        justifyContent: 'center',
        alignItems: 'center'
    },
    iconContainer: {
        width: 50,
        height: 50,
        borderRadius: 50,
        backgroundColor: '#d9d7d7',
        alignItems: 'center',
        justifyContent: 'center'
    },
    headerTitle: {
        fontSize: 20,
        fontWeight: '600',
        color: 'black',
        margin: 10
    },
    description: {
        fontSize: 14,
        color: 'black',
        textAlign: 'center',
        marginHorizontal: 40
    }
})