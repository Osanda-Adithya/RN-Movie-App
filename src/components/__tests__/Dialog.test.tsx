import React from 'react';
import Dialog from '../Dialog';
import { render } from '@testing-library/react-native'

const component = (
    <Dialog
        title={''}
        description={''}
    />
)

describe('render successfully', () => {
    it('render snapshot', () => {
        const container = render(component);
        expect(container.toJSON()).toMatchSnapshot();
    })
})