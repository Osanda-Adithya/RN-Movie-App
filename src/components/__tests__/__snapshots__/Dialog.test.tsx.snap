// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render successfully render snapshot 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "rgba(4, 36, 68, 0.8)",
      "height": 1334,
      "justifyContent": "center",
      "position": "absolute",
      "width": 750,
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "white",
        "borderRadius": 10,
        "height": 200,
        "justifyContent": "center",
        "width": 700,
      }
    }
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "#d9d7d7",
          "borderRadius": 50,
          "height": 50,
          "justifyContent": "center",
          "width": 50,
        }
      }
    >
      <Text
        allowFontScaling={false}
        selectable={false}
        style={
          Array [
            Object {
              "color": "red",
              "fontSize": 26,
            },
            undefined,
            Object {
              "fontFamily": "Ionicons",
              "fontStyle": "normal",
              "fontWeight": "normal",
            },
            Object {},
          ]
        }
      >
        ï–¹
      </Text>
    </View>
    <Text
      style={
        Object {
          "color": "black",
          "fontSize": 20,
          "fontWeight": "600",
          "margin": 10,
        }
      }
    />
    <Text
      style={
        Object {
          "color": "black",
          "fontSize": 14,
          "marginHorizontal": 40,
          "textAlign": "center",
        }
      }
    />
  </View>
</View>
`;
